### PR TITLE
add support for enabling fallback in sanity check to consider lib64 equivalent for seemingly missing libraries

### PR DESCRIPTION
**Pull request by boegel**
This is an alternate implementation of the proposed change by @torbjoernk in #1405, as a fix for #1404.

I prefer keeping this optional, since having the fallback enabled by default may result in the sanity check passing where it perhaps shouldn't (although maybe I'm being too careful here, I'm not sure...).

This is fueled by the issues that @pmatousu has been reporting for failing sanity checks on SUSE systems, cfr. https://github.com/easybuilders/easybuild-easyblocks/issues/1400 and https://github.com/easybuilders/easybuild-easyconfigs/issues/6165 .

### CODE REVIEW

**Comment by vanzod** [comment]
Very useful addition. I would even push it forward an make it two-ways, so that `lib <--> lib64`. This would simplify the sanity check in easyconfig files like the ones in [1] and [2].
Ideally I would like having it as a default behavior, despite the risks of breaking installations that rely on library directory names that are hardcoded in the easyblocks. In such cases it is a "bug" that should be addressed anyway.

[1] https://github.com/easybuilders/easybuild-easyconfigs/pull/4722
[2] https://github.com/easybuilders/easybuild-easyconfigs/pull/5897

**Comment by migueldiascosta** [comment]
As a more general solution, perhaps for later, wondering if wouldn't be better to put symlinks to all libs found inside `installdir` in a single folder and only check/expose that one...

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -2156,15 +2171,15 @@ def _sanity_check_step(self, custom_paths=None, custom_commands=None, extension=
...
-                        found = True
-                        break
-                    else:
-                        self.log.debug("Could not find %s %s in %s" % (typ, name, self.installdir))
+
+                found = check_path(xs, typ, check_fn)
+
+                # for library files in lib/, also consider fallback to lib64/ equivalent
+                if not found and build_option('lib64_fallback_sanity_check'):
+                    if all(x.startswith('lib/') for x in xs):
```
Comment (wpoely86) [inline easybuild/framework/easyblock.py]: I don't understand this `all`? why do all paths have to start with `lib/`?

Comment (wpoely86) [reply easybuild/framework/easyblock.py]: a wait, xs is a tuple which we do OR?

Comment (boegel) [reply easybuild/framework/easyblock.py]: Yes, `xs` is one entry from e.g. the `files` list from `sanity_check_paths` can be a tuple (and if it's just a string, we make it a tuple with a single element higher up).

**Comment by boegel** [comment]
@vanzod Reverse (also considering `lib/libfoo.a` when `lib64/libfoo.a` is not found) is now also implemented.

In addition, I enabled the lib64 fallback in the sanity check by default; it can be disabled if desired via `--disable-lib64-fallback-sanity-check`.

**Pull request merged**